cassandraprotocol/client/client_test.go

```diff
@@ -0,0 +1,247 @@
+package client
+
+import (
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/compression"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/frame"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/message"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var compressors = map[string]*frame.Codec{
+	"LZ4":    frame.NewCodec(frame.WithCompressor(&compression.Lz4Compressor{})),
+	"SNAPPY": frame.NewCodec(frame.WithCompressor(&compression.SnappyCompressor{})),
+	"NONE":   frame.NewCodec(),
+}
+
+// This test requires a remote server listening on localhost:9042 without authentication.
+func TestRemoteServerNoAuth(t *testing.T) {
```

Comment:
These tests will have to be removed because they require an external server, otherwise they can fail. But I included them for now to illustrate how to use this new client-server API.

Reply:
maybe add a check on a test flag or env variable so that by default `go test` doesn't run this

Reply:
Same comment for other tests that require an external server

Reply:
Very good idea, will do! 👍